src/deps_deploy/deps_deploy.clj

```diff
@@ -82,6 +88,62 @@
 (defn- artifact [{:keys [group-id artifact-id version]}]
   (str group-id "/" artifact-id "-" version))
 
+(defn- read-edn-files
+  "Given as options map, use tools.deps.alpha to read and merge the
+  applicable `deps.edn` files, from depstar"
+  [{:keys [repro] :or {repro true}}]
+  (let [{:keys [root-edn user-edn project-edn]} (t/find-edn-maps)]
+    (t/merge-edns (if repro
+                    [root-edn project-edn]
+                    [root-edn user-edn project-edn]))))
+
+(defn- set-settings-builder
+  [^DefaultMavenSettingsBuilder default-builder settings-builder]
+  (doto (.. default-builder getClass (getDeclaredField "settingsBuilder"))
+    (.setAccessible true)
+    (.set default-builder settings-builder)))
+
+(defn- get-settings
+  ^Settings []
+  (.buildSettings
+    (doto (DefaultMavenSettingsBuilder.)
+      (set-settings-builder (.newInstance (DefaultSettingsBuilderFactory.))))))
+
+(defn- preprocess-options
+  "Given an options hash map, if any of the values are keywords, look them
+  up as alias values from the full basis (including user `deps.edn`).
+  :installer is the only option that is expected to have a keyword value
+  so we skip the lookup for that.
+  Typically the value of the :repository option is a hashmap, if it is string,
+  that string is used as the key to get the repoistory hashmap from the :mvn/repos hashmap from the full basis.
+  Code to read ~/.m2/settings.xml and get auth credentials from clojure.tools.deps.alpha
+  Based on same fn in depstar"
+  [options]
+  (let [edn-files (read-edn-files {:repro false})
+        aliases   (:aliases edn-files)
+        mvn-repos (:mvn/repos edn-files)]
+    (reduce-kv (fn [opts k v]
+                 (cond
+                   (and (not= :installer k) (keyword? v))
+                     (if (contains? aliases v)
+                       (assoc opts k (get aliases v))
+                       (do
+                         (println k "has value" v "which is an unknown alias")
+                         opts))
+                     (and (= :repository k) (string? v))
+                       (if (contains? mvn-repos v)
+                         (let [settings (get-settings)
```

Comment:
I would prefer that the parsing of settings are done outside of `preprocess-options` and that `repo-settings` are passed as a parameter to `preprocess-options`, if that's possible.
This way, it would be possible to write tests for `preprocess-options` (which by looking at the complexity of the function, might just be called for).

So I'd believe `preprocess-options` should be defined as something like:

`(defn- preprocess-options [options repo-settings] ...)`

Reply:
> This way, it would be possible to write tests for preprocess-options (which by looking at the complexity of the function, might just be called for).

Yes, point taken.  I'll look into refactoring this code based on your comments.

Reply:
> repo-settings are passed as a parameter to preprocess-options, if that's possible.

I don't see any easy way to do that.  Obtaining `repo-settings` requires knowing the value of the exec-arg `:repository` is not a map, but a string, and the string itself.  That isn't known until preprocess-options reduces down the options.
I am able to refactor/move the code that obtains the repo settings into `(get-repo-settings repo mvn-repos)` as you will see (shortly)

Reply:
Based on your comments, I attempted to seperate concerns, moving the repo access code into `get-repo-settings`.
Now `preprocess-options` is all about iterating though the list of options, and replacing "magic" option values.